src/storage/txn/store.rs

```diff
@@ -116,39 +137,40 @@ impl<S: Snapshot> SnapshotStore<S> {
     }
 }
 
+impl<S: Snapshot> SnapshotStore<S> {
+    pub fn new(
+        snapshot: S,
+        start_ts: u64,
+        isolation_level: IsolationLevel,
+        fill_cache: bool,
+    ) -> Self {
+        SnapshotStore {
+            snapshot,
+            start_ts,
+            isolation_level,
+            fill_cache,
+        }
+    }
+}
+
 pub struct StoreScanner<S: Snapshot> {
     forward_scanner: Option<ForwardScanner<S>>,
```

Comment:
Maybe we only need one scanner? They have the same APIs and there behaviors are different and they never exist at the same time

Reply:
Changes are too much. Let's change it in another PR. @MyonKeminta